cognite/neat/_rules/exporters/_rules2dms.py

```diff
@@ -306,50 +307,38 @@ def _categorize_items_for_upload(
 
             cdf_item = cdf_item_by_id.get(loader.get_id(item))
             if cdf_item is None:
-                to_create.append(item)
-            elif is_redeploying:
-                to_update.append(item)
-                to_delete.append(cdf_item)
+                categorized.to_create.append(item)
+            elif is_redeploying or self.existing == "recreate":
+                if loader.has_data(cdf_item) and not self.drop_data:
+                    categorized.to_skip.append(cdf_item)
+                else:
+                    categorized.to_delete.append(cdf_item.as_write())
+                    categorized.to_create.append(item)
```

Comment:
should item also be `.as_write` ?

Reply:
no, the item is the local one which is already an write item.